### PR TITLE
EPD-568 initial work on ruby SDK

### PR DESCRIPTION
**Pull request by sameerank**
[EPD-568](https://linear.app/eppo/issue/EPD-568/[sdk]-ruby-sdk)

All the commands are in the rakefile (which is the ruby equivalent of a makefile). In order to have all the dev dependencies initially (which includes rake, which would enable you to run the rakefile) to use the file, you'll need to do an installation first without the rakefile.

```
gem build eppo-server-sdk.gemspec
gem install eppo-server-sdk-0.0.0.gem --dev
```

Afterwards, you can run `rake test`, which will re-run the build-install and download the test data before running rspec.

To just download the test data, you can run `rake test-data`

### CODE REVIEW

**Review by jonathan-fulton — APPROVED** [review]
I've never really done vanilla Ruby (mostly Rails) but I think overall this looks good.  Definitely appreciate the tests.

File: lib/client.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      end
+
+      allocation = experiment_config.allocations[matched_rule.allocation_key]
+      unless in_experiment_sample?(
+        subject_key,
+        flag_or_experiment_key,
+        experiment_config.subject_shards,
+        allocation.percent_exposure
+      )
+        EppoClient.logger('out').info(
```
[the remark this answers is not in the document]
Comment (sameerank) [reply lib/client.rb]: In the [node version of getAssignment](https://github.com/Eppo-exp/node-server-sdk/blob/main/src/client/eppo-client.ts#L41) we don't do any logging unless it errors, and [in the python version](https://github.com/Eppo-exp/python-sdk/blob/main/eppo_client/client.py#L34), we are logging at the info level, which is the behavior I duplicated here.

BUT I like the idea of logging at the debug level, so I just did the following: 1) changed these to `EppoClient.logger.debug`, 2) set the default log level to `Logger::INFO`, and 3) added an argument to `get_assignment` to allow the end user to change the log level to `Logger::DEBUG` if they want to see these errors

Would the other SDKs would need to be updated to match this?

File: eppo-server-sdk.gemspec [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+Gem::Specification.new do |s|
+  s.name        = 'eppo-server-sdk'
+  s.version     = '0.0.1'
```
[the remark this answers is not in the document]
Comment (sameerank) [reply eppo-server-sdk.gemspec:5]: Oh I didn't realize that I could use `parse_gemspec` in the Rakefile. Now the version is only in one place: the gemspec file

[the remark this answers is not in the document]
Comment (sameerank) [reply eppo-server-sdk.gemspec:5]: I also updated the `README.md` to include some pre-installation instructions to make the Rakefile useable (the developer would need to have installed `rspec` and `parse_gemspec`). I'm assuming it's okay for end users to find that in the readme if they're peaking at the source code

**Pull request merged**